### ISSUE TITLE
Subtitles

### ISSUE THREAD

**Issue by joschahenningsen** [joschahenningsen/TUM-Live#796]
This feature tracks the progress of implementing auto-generated subtitles and a related search functionality.

Related: https://github.com/TUM-Dev/TUM-Live-Voice-Service

**Comment by joschahenningsen** [comment]
Status: We already have a functioning micro-service that does the transcribing. The back-end functionality to call the micro-service and handle the data is still missing. Also the front-end wasn't adapted yet to support .srt subtitles but this should be easy. 

We might want to consider using whisper instead of vosk in the future (thanks to the hackatum team "Wordless"!) but this can be tracked in a separate issue at https://github.com/TUM-Dev/TUM-Live-Voice-Service

Drawbacks: whisper needs much more resources so we might want to make it configurable which service to use.

**Comment by simon300000** [comment]
(commenting here because it maybe more a management problem)

Is it possible to distribute the workload when using whisper?

Some students have gaming computer/laptop with dedicated GPU, is it possible that we can “crowdfund” computing resources? So volunteer opens an executable which pulls audio from Tum-live and upload the SRT file.

To motivate people we might have some kind of leaderboard or even gifts at end of each semester, I think many would like to help.

**Comment by CommanderStorm** [comment]
Thanks for the offer, but this won't be necessary. (Furthermore, I am not convinced that this would be an option in terms of data-protection and reliability)
We have enough compute (the BBB nodes are currently not as busy as they used to 😉). This would be via CPUs, which are slower than real-time (not good, but manageable as lectures are not streamed 24/7).
RBG is currently also looking into buying GPUs.

This will definitely happen, but it will possibly take a few weeks ;) 
@joschahenningsen Correct me if I am distributing outdated info 😉

**Comment by joschahenningsen** [comment]
> This will definitely happen, but it will possibly take a few weeks ;)

Exactly, I'll get a few powerful servers but it will take some time as we're a little understaffed right now
